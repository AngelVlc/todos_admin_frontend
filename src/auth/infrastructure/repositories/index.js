export * from './RefreshTokensRepository';
export * from './UsersRepository';
