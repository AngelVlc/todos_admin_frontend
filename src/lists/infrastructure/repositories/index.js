export * from "./ListsRepository";
export * from "./ListItemsRepository";
