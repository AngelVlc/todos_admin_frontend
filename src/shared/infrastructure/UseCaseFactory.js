export class UseCaseFactory {
  get(useCase) {
    return useCase.create();
  }
}
