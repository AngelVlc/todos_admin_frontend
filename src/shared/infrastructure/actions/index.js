export * from './requests';
export * from './login';
