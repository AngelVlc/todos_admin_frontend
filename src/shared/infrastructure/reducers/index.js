export * from './loginReducer';
export * from './requestsReducer';
