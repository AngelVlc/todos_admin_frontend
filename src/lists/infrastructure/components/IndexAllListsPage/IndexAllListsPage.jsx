import React, { useContext } from "react";
import { AppContext } from "../../../../shared/infrastructure/contexts";
import { Breadcrumb } from "../../../../shared/infrastructure/components/Breadcrumb";
import { YesNoButtons } from "../../../../shared/infrastructure/components/YesNoButtons";
import { useHistory } from "react-router-dom";
import { IndexAllListsUseCase } from "../../../application";

export const IndexAllListsPage = () => {
  const { useCaseFactory } = useContext(AppContext);
  let history = useHistory();

  const indexAllDocs = async () => {
    const indexAllUseCase = useCaseFactory.get(IndexAllListsUseCase);

    if (await indexAllUseCase.execute()) {
      history.push("/");
    }
  };

  return (
    <div className="container">
      <Breadcrumb
        items={[{ url: "/index-all-lists", text: "Index All Lists" }]}
      />
      <h3 className="title">Index All Lists</h3>
      <YesNoButtons onYes={indexAllDocs} onNo={() => history.push("/")} />
    </div>
  );
};
