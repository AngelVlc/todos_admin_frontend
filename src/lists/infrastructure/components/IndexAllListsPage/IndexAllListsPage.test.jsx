import { render, cleanup, fireEvent, waitFor } from "@testing-library/react";
import { IndexAllListsPage } from "./IndexAllListsPage";
import { AppContext } from "../../../../shared/infrastructure/contexts";
import { MemoryRouter, Route } from "react-router-dom";
import { act } from "react-dom/test-utils";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const mockedIndexAllUseCase = {
  execute: jest.fn(),
};


const renderWithContextAndRouter = () => {
  const useCaseFactory = {
    get: () => mockedIndexAllUseCase,
  };

  const context = { auth: { info: {} }, useCaseFactory };

  return {
    ...render(
      <AppContext.Provider value={context}>
        <MemoryRouter initialEntries={['/']}>
          <Route>
            <IndexAllListsPage />
          </Route>
        </MemoryRouter>
      </AppContext.Provider>
    ),
  };
};

afterEach(cleanup);

describe("IndexAllListsPage", () => {
  it("should match the snapshot", async () => {
    let fragment;
    await act(async () => {
      const { asFragment } = renderWithContextAndRouter();
      fragment = asFragment;
    });

    expect(fragment()).toMatchSnapshot();
  });

  it("when click on cancel should cancel the indexing", async () => {
    const container = renderWithContextAndRouter();

    await waitFor(() => {
      fireEvent.click(container.getByTestId("no"));
    });

    expect(mockHistoryPush).toHaveBeenCalled();
    expect(mockHistoryPush.mock.calls[0][0]).toBe("/");
    mockHistoryPush.mockClear();
  });

  it("when click on yes should index all the List", async () => {
    const container = renderWithContextAndRouter();

    mockedIndexAllUseCase.execute.mockResolvedValue(true);

    await waitFor(() => {
      fireEvent.click(container.getByTestId("yes"));
    });

    expect(mockedIndexAllUseCase.execute).toHaveBeenCalled();
    expect(mockedIndexAllUseCase.execute.mock.calls[0][0]).toBe(undefined);
    expect(mockHistoryPush).toHaveBeenCalled();
    expect(mockHistoryPush.mock.calls[0][0]).toBe("/");
    mockHistoryPush.mockClear();
  });
});
