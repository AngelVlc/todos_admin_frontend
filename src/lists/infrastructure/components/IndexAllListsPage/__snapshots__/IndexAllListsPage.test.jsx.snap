// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexAllListsPage should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <nav
      aria-label="breadcrumbs"
      class="breadcrumb"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="is-active"
        >
          <a
            aria-current="page"
            href="/index-all-lists"
          >
            Index All Lists
          </a>
        </li>
      </ul>
    </nav>
    <h3
      class="title"
    >
      Index All Lists
    </h3>
    <div
      class="buttons"
    >
      <button
        class="button is-danger"
        data-testid="yes"
      >
        YES
      </button>
      <button
        class="button"
        data-testid="no"
      >
        NO
      </button>
    </div>
  </div>
</DocumentFragment>
`;
